### PR TITLE
(PXP-7983): Enable authentication with access token for GET /token endpoint

### PR DESCRIPTION
**Pull request by johnfrancismccann**
Jira Ticket: [PXP-7983](https://ctds-planx.atlassian.net/browse/PXP-7983)

### New Features
- Enable authentication with access token for `GET /token` endpoint

### CODE REVIEW

File: tests/app_test.py [hunk 1]
```diff
@@ -98,38 +98,64 @@ def test_connected_endpoint(client, test_user, db_session, auth_header):
     assert res.status_code == 200
 
 
-def test_token_endpoint(client, test_user, db_session):
+def test_token_endpoint_with_default_idp(client, test_user, db_session, auth_header):
```
Comment (johnfrancismccann) [inline tests/app_test.py:101]: All of the `client.get` requests originally in `test_token_endpoint` seemed to correspond to the same http request on the backend (e.g. when using a debugger, `hasattr(flask.g, "user")` in [`wts/auth.py`](https://github.com/uc-cdis/workspace-token-service/blob/master/wts/auth.py) evaluated to `False` for the first `client.get`, but then evaluated to `True` for each subsequent `client.get`), so in order to isolate `test_token_endpoint_without_auth_header` so it really doesn't have an auth header, I had to put each `client.get` request in its own test.

File: wts/auth_plugins/base.py [hunk 2]
```diff
@@ -13,4 +20,15 @@ def find_user(self):
...
+        default_oauth_client = get_oauth_client(idp="default")
+        flask.current_app.config["OIDC_ISSUER"] = default_oauth_client.metadata[
+            "api_base_url"
+        ].rstrip("/")
+        try:
+            user = current_user
+            return User(userid=user.id, username=user.username)
+        except:
+            #  allow other plugins a chance to authenticate
+            return None
```
Comment (johnfrancismccann) [inline wts/auth_plugins/base.py:34]: One thing to note is that for this [plugin to get used](https://github.com/uc-cdis/workspace-token-service/blob/081300e795382b6c9e7d17bb083398489e40a278/wts/api.py#L54) when the `AUTH_PLUGINS` env variable exists, `default` must appear in `AUTH_PLUGINS` (e.g. `AUTH_PLUGINS=default,k8s`).

Right now, cloud automation [hard codes](https://github.com/uc-cdis/cloud-automation/blob/c445e42ad69a8a65eb8846e19cb2d09869f71fd6/kube/services/wts/wts-deploy.yaml#L56) this to `k8s`. Maybe, if necessary, we could configure `AUTH_PLUGINS` per commons using a secret, but I probably think just updating `wts-deploy.yaml` with `AUTH_PLUGINS=default,k8s` will work for now.

Comment (Avantol13) [reply wts/auth_plugins/base.py:34]: Can you create an accompanying PR to do that in cloud auto? Will that be a breaking change? e.g. if cloud auto gets updated to that but this new code isn't deployed will it still work okay?

**Comment by Avantol13** [comment]
is it possible to make `find_user` a bit smarter for prioritizing which configured plugins to use? I'm thinking we could simply check if there's an "Authorization" header and then prefer the "default"/access token plugin, so that in the workspaces/k8s pod case, I believe there's no "Authorization" header (will need to verify) so we could maybe make this a bit more efficient by not even trying `find_user` for the "default"/access token plugin in that case and go directly to the k8s one (and that will probably save time?) 

Does that make sense? Maybe it's too much optimization and just something to keep in mind for the future.. iterating over options without some smart prioritization just has a bit of a code smell that I don't like

File: wts/auth_plugins/base.py (hunk 2 above)
Comment (johnfrancismccann) [reply wts/auth_plugins/base.py:34]: I forgot about that. Updating `wts-deploy.yaml` with `AUTH_PLUGINS=default,k8s` would actually be a breaking change because in `master`, the `"test"` `User` is automatically returned so the Kubernetes plugin [wouldn't get a chance to run](https://github.com/uc-cdis/workspace-token-service/blob/081300e795382b6c9e7d17bb083398489e40a278/wts/auth_plugins/__init__.py#L19). Also, if a refresh token did exist in the database for a `"test"` user, anyone could get a `"test"` access token.

**Comment by johnfrancismccann** [comment]
Initially I thought optimizing how `find_user` runs the plugins would make the code a bit more complex, but I decided to try it out in a [branch of this branch](https://github.com/uc-cdis/workspace-token-service/compare/feat/authenticate-with-access-token...feat/authenticate-with-access-token-optimize-plugin-choice). As it turns out, the optimized code is more concise. I think it also has the additional benefit of not being coupled to the `AUTH_PLUGINS` environment variable set by `cloud-automation`.

Also, when testing the “Export to Workspace” flow for the branch of this branch, `auth_header: None` was printed for the requests made from Fuse.

File: wts/auth_plugins/__init__.py [hunk 3]
```diff
@@ -3,17 +3,7 @@
...
-
-
 def find_user():
-    for plugin in flask.current_app.auth_plugins:
-        user = plugin.find_user()
-        if user:
-            return user
+    if flask.request.headers.get("Authorization"):
+        return DefaultPlugin().find_user()
+    return K8SPlugin().find_user()
```
Comment (Avantol13) [inline wts/auth_plugins/__init__.py:9]: we may want to keep the configuration and plugin setup but change it to only handle situations where a token is not provided. e.g. some alternative auth provider besides fence (in this case, the only one we'd have right now is k8s but maaaybe there's a case to be made for something else in the future?) maybe this is YAGNI, but the codes already basically there, we could just always setup "default" and allow cfg setup of "k8s". Then here we iterate over the plugins AFTER we check if there's an Authorization header.
